File: examples/expo-native-module/modules/device-metadata/android/src/main/java/expo/modules/devicemetadata/DeviceMetadataModule.kt
package expo.modules.devicemetadata

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DeviceMetadataModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("DeviceMetadata")

    Events("onChange")

    Constant("PI") {
      Math.PI
    }

    Function("hello") {
      "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { value: String ->
      sendEvent("onChange", mapOf(
        "value" to value
      ))
    }

    View(DeviceMetadataView::class) {
    }
  }
}

File: examples/expo-native-module/modules/device-metadata/ios/DeviceMetadataModule.swift
import ExpoModulesCore

public class DeviceMetadataModule: Module {
  public func definition() -> ModuleDefinition {
    Name("DeviceMetadata")

    Events("onChange")

    Constant("PI") {
      Double.pi
    }

    Function("hello") {
      return "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { (value: String) in
      self.sendEvent("onChange", [
        "value": value
      ])
    }

    View(DeviceMetadataView.self) {
    }
  }
}
